build/ci/wait_for_k8s_ready.sh

```diff
@@ -0,0 +1,16 @@
+#!/bin/bash -xe
+set +o pipefail
+
+echo "Wait for all the pods to be in running state"
+while [ "$(kubectl get pod -A | grep 0/ | grep -iv name | wc -l)" -gt 0 ]; do
+  echo Some pods did not start thier containers
+  kubectl get pod -A | grep 0/ | grep -iv name
+  sleep 5
+done
+while [ "$(kubectl get pod -A | grep -iv running | grep -iv name | wc -l)" -gt 0 ]; do
+  echo Some pods are still not in running state
+  kubectl get pod -A | grep -iv running | grep -iv name
+  sleep 5
+done
```

Comment:
1. do we really need to check with `0/` separately?
2. shouldn't we filter with `-l csi`, to be on the safe side?
3. please reduce code duplications as much as possible

Reply:
1. because there are pods that are in running mode but still in 0/1 containers that up

Reply:
1. theoretically, could they be in e.g. 1/2 containers that are up?

Reply:
I didn't face a situation that it happened and the pod was in a Running state(it was in ERROR/CONTAINERCREATING state) but I can't tell for sure

Reply:
could we check only that the expected valid state exists, without assuming which invalid states exist or not exist?

Reply:
done